### PR TITLE
Makes dead metabolization work properly

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

So, currently, on_mob_dead() metabolization is very barebones. It can't overdose and doesn't work when your liver is failing or you simply don't have one. It's been completely neglected and isn't up-to-date on all the fixes done to on_mob_life() and general metabolization.

This PR moves on_mob_dead() into the general metabolization chain and gives it it's own handling proc in the life cycle called handle_dead_metabolization() which simply calls metabolize() with dead and liverless. Sadly your liver doesn't work when you're dead, though self consuming reagents will still pass through due to my previous fixes to reagent processing.
## Why It's Good For The Game

Not having desynchronized updates on the metabolization procs is good for the sake of code quality, readability and futureproofing.

Making a proc do what it's supposed to is also a pretty good change.
## Changelog
:cl:
fix: Reagents metabolize properly when dead. (If they're supposed to.)
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -821,3 +822,12 @@
 		return
 
 	heart.beating = !status
+
+//////////////////////
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/life.dm:826]: don't do this. it ruins auto doc and looks horrible

File: code/modules/reagents/chemistry/holder.dm [hunk 2]
```diff
@@ -749,15 +749,15 @@
...
+/datum/reagents/proc/metabolize_reagent(mob/living/carbon/owner, datum/reagent/reagent, delta_time, times_fired, can_overdose = FALSE, liverless = FALSE, dead = FALSE)
 	var/need_mob_update = FALSE
 	if(QDELETED(reagent.holder))
 		return FALSE
 
 	if(!owner)
 		owner = reagent.holder.my_atom
 
-	if(owner && reagent)
+	if(owner && reagent && !(dead && !(reagent.chemical_flags & REAGENT_DEAD_PROCESS)))
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:760]: Did you mean `||` here instead of `&&`?

File: code/modules/mob/living/carbon/life.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/mob/living/carbon/life.dm:826]: Actually document what this does and when it is called

File: code/modules/reagents/chemistry/holder.dm (hunk 2 above)
Comment (RikuTheKiller) [reply code/modules/reagents/chemistry/holder.dm]: Yep, turns out my brain didn't realize the obvious way to do this.

File: code/modules/mob/living/carbon/life.dm (hunk 1 above)
Comment (RikuTheKiller) [reply code/modules/mob/living/carbon/life.dm]: Done, I also moved it to the area with all the other handler procs.

**Pull request merged**